beacon-chain/p2p/broadcaster.go

```diff
@@ -41,12 +47,32 @@ func (s *Service) Broadcast(ctx context.Context, msg proto.Message) error {
 func (s *Service) BroadcastAttestation(ctx context.Context, subnet uint64, att *eth.Attestation) error {
 	ctx, span := trace.StartSpan(ctx, "p2p.BroadcastAttestation")
 	defer span.End()
+	oneEpoch := time.Duration(1*params.BeaconConfig().SlotsPerEpoch*params.BeaconConfig().SecondsPerSlot) * time.Second
+	ctx, cancel := context.WithTimeout(ctx, oneEpoch)
+	defer cancel()
 	forkDigest, err := s.forkDigest()
 	if err != nil {
 		err := errors.Wrap(err, "could not retrieve fork digest")
 		traceutil.AnnotateError(span, err)
 		return err
 	}
+	// Ensure we have peers with this subnet.
+	if !s.hasPeerWithSubnet(subnet) {
+		for {
+			if err := ctx.Err(); err != nil {
+				return err
+			}
+			ok, err := s.FindPeersWithSubnet(ctx, subnet)
```

Comment:
A bit worried for constantly calling this for an epoch. This method can get expensive as it does constant network lookups. Can we bound this ? If a peer has issues finding peers because, ex: behind a NAT, etc this will be spamming network lookups which can lead to a large increases in CPU and memory. Maybe after a certain amount of tries, we just return an error ? that way at least the user knows something is wrong